### DISCUSSION TITLE
Contributing to brew and learning its internals

### DISCUSSION THREAD

**Discussion by AmirNaghibi** [Homebrew/discussions#3901] [Everyday usage]
Are there any architecture diagrams or resources to learn about internals of brew? 

For example, if there is a new developer joining Homebrew and wants to contribute, what resources do you recommend to get an overall understanding of the components and services that are involved in brew? (beside the `brew audit` command that was on the README's Contributing section)

**Comment by SMillerDev** [comment]
> Are there any architecture diagrams or resources to learn about internals of brew?

docs.brew.she has all the documentation for Homebrew

> For example, if there is a new developer joining Homebrew and wants to contribute, what resources do you recommend to get an overall understanding of the components and services that are involved in brew?

I would recommend reading the part that people want to change and writing tests to ensure it does what they want. No need for an in depth understanding of all things Homebrew.
